### PR TITLE
MP-8380 - Creation of Nox Type "Guid"

### PR DESCRIPTION
**Pull request by danield-ap**

### CODE REVIEW

File: src/Nox.Types/Types/Guid/Guid.cs [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+        }
+
+        return newObject;
+    }
+
+    /// <summary>
+    /// Initialize a new instance of <see cref="Guid"/> with an autogenerated Guid in its <see cref="ValueObject{T,TValueObject}.Value"/>.
+    /// </summary>
+    /// <returns>A new <see cref="Guid"/> object.</returns>
+    public static Guid NewGuid() => new() { Value = System.Guid.NewGuid() };
```
Comment (gjorgji-grgovski-authoritypartners) [inline src/Nox.Types/Types/Guid/Guid.cs:47]: Can we also have an Empty Guid and test for it?

File: tests/Nox.Types.Tests/Types/Guid/GuidTests.cs [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+        var guid = Guid.From(textGuid);
+
+        guid.Should().NotBeNull();
+        guid.Value.Should().NotBeEmpty();
+        guid.ToString().Should().Be(textGuid);
+    }
+
+    [Fact]
+    public void Guid_ToString_ReturnSameGuid()
```
Comment (gjorgji-grgovski-authoritypartners) [inline tests/Nox.Types.Tests/Types/Guid/GuidTests.cs:21]: This test is doing the same as the first one, can you merge them?

**Review by gjorgji-grgovski-authoritypartners — CHANGES_REQUESTED** [review]

File: tests/Nox.Types.Tests/Types/Guid/GuidTests.cs (hunk 2 above)
Comment (danield-ap) [reply tests/Nox.Types.Tests/Types/Guid/GuidTests.cs]: Tests merged.

File: src/Nox.Types/Types/Guid/Guid.cs (hunk 1 above)
Comment (danield-ap) [reply src/Nox.Types/Types/Guid/Guid.cs:52]: `Guid.Empty` created and also added it to validation so we don't allow creating an empty Guid.

File: src/Nox.Types/Types/Guid/Guid.cs [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+namespace Nox.Types;
+
+/// <summary>
+/// Represents a Nox <see cref="Guid"/> type and value object.
+/// </summary>
+public sealed class Guid : ValueObject<System.Guid, Guid>
+{
+    /// <summary>
+    /// A instance of the <see cref="Guid"/> object whose Value is all zeros.
+    /// </summary>
```
Comment (rochar) [inline src/Nox.Types/Types/Guid/Guid.cs:12]: Create it from Sytem.Guid.Empty no need for string hardcoded

**Review by rochar — CHANGES_REQUESTED** [review]

File: src/Nox.Types/Types/Guid/Guid.cs (hunk 3 above)
Comment (danield-ap) [reply src/Nox.Types/Types/Guid/Guid.cs:12]: Updated to create it from `System.Guid.Empty`.

**Review by gjorgji-grgovski-authoritypartners — APPROVED** [review]

**Review by rochar — APPROVED** [review]

**Pull request merged**